src/mapgen_v6.cpp

```diff
@@ -847,21 +849,20 @@ void MapgenV6::flowMud(s16 &mudflow_minpos, s16 &mudflow_maxpos)
 				}
 
 				// Drop mud on side
-				for (u32 di = 0; di < 4; di++) {
-					v3s16 dirp = dirs4[di];
+				for (const v3s16 &dirp : dirs4) {
```

Comment:
yes, for range-based loop iterate in incremental order, if you want other order you cannot use it